Fixed support beams to only be placeable on beams at least three blocks high

DavidTriphon opened the pull request:
What it says on the tin.

![2020-02-13_00 28 40](https://user-images.githubusercontent.com/20526831/74415380-1eab6a80-4df8-11ea-8b2a-bc66cb7023ae.png)

DavidTriphon committed:
restricted horizontal beam placement to only work when the vertical beam is three high. Older beams that were already placed on shorter vertical beams will not break randomly. They are effectively 'grandfathered in'.

Fr4gtastic commented:
To be clear — is it only three błocks or at least three błocks?

DavidTriphon commented:
At least. It only checks the two blocks beneath the attachment block. Beyond that it doesn’t care if it is solid or support.

DavidTriphon commented:
The picture shows beams placed on 3, 4, and 5 height beams but not beneath that.

Fr4gtastic commented:
I see, thanks for the clarification.

DisasterMoo commented on src/main/java/net/dries007/tfc/objects/blocks/wood/BlockSupport.java:
```diff
@@ -292,10 +292,8 @@ public boolean canPlaceBlockOnSide(@Nonnull World world, @Nonnull BlockPos pos,
-            }
+            if (!isConnectable(world, pos, side.getOpposite())) return false;
+            if (!isThreeTall(world, pos.offset(side.getOpposite()))) return false;
             int distance = getHorizontalDistance(side, world, pos);
```
This is only checking on one side. Better to implement that logic on `getHorizontalDistance` and only return a value if both sides are at least three tall.

DavidTriphon committed:
Moved three tall logic to get distance method per Moo's suggestion. Also solved issue with checking both sides.

DavidTriphon commented:
Fixed 3 tall check to work on both sides and moved 3 tall check to the `getHorizontalDistance` method.

![2020-02-13_11 04 20](https://user-images.githubusercontent.com/20526831/74469083-ceacc200-4e50-11ea-9068-f8913cb0747d.png)
